clang/lib/Driver/ToolChains/Darwin.cpp

```diff
@@ -1645,14 +1645,14 @@ void DarwinClang::AddLinkRuntimeLibArgs(const ArgList &Args,
     CmdArgs.push_back("-lSystem");
 
   // Select the dynamic runtime library and the target specific static library.
-  if (isTargetIOSBased()) {
-    // If we are compiling as iOS / simulator, don't attempt to link libgcc_s.1,
-    // it never went into the SDK.
-    // Linking against libgcc_s.1 isn't needed for iOS 5.0+
-    if (isIPhoneOSVersionLT(5, 0) && !isTargetIOSSimulator() &&
-        getTriple().getArch() != llvm::Triple::aarch64)
-      CmdArgs.push_back("-lgcc_s.1");
-  }
+  // If we are compiling as iOS / simulator, don't attempt to link libgcc_s.1,
```

Comment:
I suggest consolidating those comments into one for explaining when libgcc is used, basically:
* below macOS 10.6
* below iOS 5.0 but device only (not simulator)
* not on arm64 since libgcc is obsoleted before its existence.

Reply:
How does this look?
```
Select the dynamic runtime library and the target specific static library.
Some old Darwin versions put builtins, libunwind, and some other stuff in
libgcc_s.1.dylib. Newer versions moved those functions to libSystem, and
made libgcc_s.1.dylib a stub. We need libgcc_s on versions before MacOS X
10.6 and iOS 5. We also make sure to never link libgcc_s on aarch64,
since libgcc_s was made obsolete before it's existence.
```